protos/sigstore_common.proto

```diff
@@ -68,6 +68,23 @@ message HashOutput {
         bytes digest = 2;
 }
 
+// SupportedAlgorithm captures the signature/hash algorithm combinations allowed
+// in the Sigstore ecosystem.
+//
+// This is modelled as a linear set as we want to provide a small number of
+// opinionated options instead of allowing every possible permutation.
+//
+// Any changes to this enum MUST be reflected in the algorithm registry.
+// See: docs/algorithm-registry.md
+enum SupportedAlgorithm {
+        SUPPORTED_ALGORITHM_UNSPECIFIED = 0;
+        ECDSA_SHA2_256_NISTP256 = 1;
+        ECDSA_SHA2_256_NISTP521 = 2;
+        ECDSA_SHA2_384_NISTP384 = 3;
+        ED25519 = 4;
+        ED25519_PH = 5;
+}
+
```

Comment:
Shall we have separate enums for different purposes? For example, `ClientSupportedAlgorithms`, `FulcioSupportedAlgorithms`, etc.?

Is ED25519 without pre-hash viable in `cosign`/`sigstore-python` or other clients?

Reply:
Hmm, maybe rather than calling these `SupportedAlgorithm`s, we can call them `KnownAlgorithm`s -- that will avoid implying that they're supported everywhere, and we can document that clients must refer to the client spec and other documentation to determine where each should be supported.

> Is ED25519 without pre-hash viable in cosign/sigstore-python or other clients?

For `hashedrekord`, no, but for other entry types (e.g. `dsse`) I think yes.

Reply:
+1 to referring to these as Known, this is related to the comment I left above too that this list does not mandate client support.

Yep, ed25519 is supported for dsse and rekord, and any type that uses [rekor's x509.NewPublicKey](https://github.com/search?q=repo%3Asigstore%2Frekor%20%22x509.NewPublicKey%22&type=code)

Reply:
Agreed. I'll change this to `KnownAlgorithm` and add some comments (and explain the markdown) that the existence of an algorithm doesn't mean that it must be supported by a given Sigstore component.

Reply:
[9b1d4ad](https://github.com/sigstore/protobuf-specs/pull/194/commits/9b1d4ad2a10c2b1e7024c3677107e59931b7b00c)

I named it `KnownSignatureAlgorithm` to distinguish between hash algorithms that also exist in the registry.